pkg/trace/writer/trace.go

```diff
@@ -206,6 +206,10 @@ func (w *TraceWriter) flush() {
 		return
 	}
 
+	defer func(start time.Time) {
+		metrics.Timing("datadog.trace_agent.trace_writer.flush_ms", time.Since(start), nil, 1)
+	}(time.Now())
```

Comment:
Same as the above comment, `time.Now()` is evaluated right here, but it should be evaluated when the defer executes, so we need to call it inside the anonymous function.

Reply:
I think you need to re-assess how this works. For example: https://play.golang.org/p/JmQJKSEwude

Reply:
Oh that's right. I didn't realize there was `time.Since` in there. So this is good 👍